src/ClpIrV1Decoder.cpp

```diff
@@ -0,0 +1,208 @@
+
+
+#include <emscripten/emscripten.h>
+#include <emscripten/val.h>
+#include <emscripten/bind.h>
+
+#include <string>
+#include <iostream>
+
+#include "ffi/ir_stream/decoding_methods.hpp"
+#include "spdlog_with_specializations.hpp"
+#include "TimestampPattern.hpp"
+#include "streaming_compression/zstd/Decompressor.hpp"
+#include "ir/LogEventDeserializer.hpp"
+#include "ffi/encoding_methods.hpp"
+
+#ifdef __cplusplus
+#define EXTERN extern "C"
+#else
+#define EXTERN
+#endif
+
+// FIXME: use macro
+enum class LOG_VERBOSITY
+{
+    NONE = 0,
+    TRACE,
+    DEBUG,
+    INFO,
+    WARN,
+    ERROR,
+    FATAL
+};
+
+const char *VERBOSITIES[] = {
```

Comment:
how about using `std::array` of `std::string_view`, and make it as a constexpr static member of the deserializer?